### PR TITLE
[#166] Validation icon swap

### LINKED ISSUE #166: Update Validation Report page to display status on all Platform Certs
Version 1.1. of the Platform Certificate specification allows for Delta Certificates to be issued for a given device (e.g. the Platform Certificate Chain). The status of the each cert in the chain must be shown in the validation report page and must indicate the status of each cert (red/green icon). Each icon should link to its corresponding cert.

To accommodate the extra icon in the platform column on the page, the attributes column can be removed. The platform column should show one icon per cert and should show a green icon if the signature and all components pass the validation. A red icon should be used if the signature or any component check failed.

### PR DESCRIPTION
**Pull request by cyrus-dev**
This pull request contains 2 main changes, the first is transferring the status text from the attributes failure to the icon specifically for platform trust chain validation. Then this removes the third column on the validation page that singles out the icons for the attribute status. 

In addition, this status is also rolled up to the summary status icon and displays the text there as well for all that have failed.  This will modify the database.  The SupplyChainValidationSummary was updated to increase the message fields length to accommodate the underlying errors rolling up into this field.

Closes #166

### CODE REVIEW

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java [hunk 1]
```diff
@@ -104,11 +104,15 @@ public SupplyChainValidationSummary validateSupplyChain(final EndorsementCredent
...
+        Map<SupplyChainValidation.ValidationType,
+                SupplyChainValidation> validationTypeMap = new HashMap<>();
         Map<PlatformCredential, SupplyChainValidation> deltaMapping = new HashMap<>();
 
         // Validate the Endorsement Credential
         if (policy.isEcValidationEnabled()) {
-            validations.add(validateEndorsementCredential(ec, acceptExpiredCerts));
+            validationTypeMap.put(SupplyChainValidation.ValidationType.ENDORSEMENT_CREDENTIAL,
+                    validateEndorsementCredential(ec, acceptExpiredCerts));
+//            validations.add(validateEndorsementCredential(ec, acceptExpiredCerts));
```
Comment (busaboy1340) [inline HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: Still need this commented out line?

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java [hunk 2]
```diff
@@ -139,7 +145,10 @@ public SupplyChainValidationSummary validateSupplyChain(final EndorsementCredent
                         platformScv = validatePcPolicy(pc, platformScv,
                                 deltaMapping, acceptExpiredCerts);
 
-                        validations.add(platformScv);
+                        validationTypeMap.put(SupplyChainValidation
+                                .ValidationType.PLATFORM_CREDENTIAL,
+                                platformScv);
+//                        validations.add(platformScv);
```
Comment (busaboy1340) [inline HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: Same as above comment.

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java [hunk 3]
```diff
@@ -155,21 +164,29 @@ public SupplyChainValidationSummary validateSupplyChain(final EndorsementCredent
...
-                    if (baseCredential == null || pc == baseCredential && !pc.isDeltaChain()) {
+                    if (baseCredential == null || pc == baseCredential) {
                         attributeScv = validatePlatformCredentialAttributes(
                             pc, device.getDeviceInfo(), ec);
-                        validations.add(attributeScv);
-                    } else {
+                        validationTypeMap.put(SupplyChainValidation
+                                .ValidationType.PLATFORM_CREDENTIAL,
+                                attributeScv);
+//                        validations.add(attributeScv);
```
Comment (busaboy1340) [inline HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: Same as above comment.

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java (hunk 1 above)
Comment (cyrus-dev) [reply HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: removed.

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java [hunk 4]
```diff
@@ -164,53 +162,30 @@ public SupplyChainValidationSummary validateSupplyChain(final EndorsementCredent
...
-                        .select(certificateManager)
-                        .byBoardSerialNumber(pc.getPlatformSerial())
-                        .getCertificates().stream().collect(Collectors.toList());
-                        Collections.sort(chainCertificates,
-                                new Comparator<PlatformCredential>() {
-                            @Override
-                            public int compare(final PlatformCredential obj1,
-                                    final PlatformCredential obj2) {
-                                return obj1.getBeginValidity()
-                                        .compareTo(obj2.getBeginValidity());
```
[the remark this answers is not in the document]
Comment (cyrus-dev) [reply HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: I'm not sure why I didn't include this in validatePcPolicy.  I feel like I came to a conclusion that I can't remember but I also can't find another change that would handle this.  And this should still be needed.

[the remark this answers is not in the document]
Comment (cyrus-dev) [reply HIRS_AttestationCA/src/main/java/hirs/attestationca/service/SupplyChainValidationServiceImpl.java]: I'm not sure why I didn't include this in validatePcPolicy. I feel like I came to a conclusion that I can't remember but I also can't find another change that would handle this. And this should still be needed.

**Pull request merged**